nengo/utils/testing.py

```diff
@@ -179,7 +179,7 @@ class WarningCatcher(object):
     def __enter__(self):
         self.catcher = warnings.catch_warnings(record=True)
         self.record = self.catcher.__enter__()
-        warnings.simplefilter('always', append=True)
+        warnings.simplefilter('always')
```

Comment:
There's now `pytest.warns`, which does the same thing as `pytest.raises` but for warnings. And it does this properly. Maybe we should switch to using that at some point. We would need to restrict ourselves to newer `pytest` though. For now, this is fine.